warehouse/query-core/src/main/java/datawave/query/predicate/ValueToAttributes.java

```diff
@@ -149,77 +162,66 @@ public String getCompositeAttributeKey(String composite, String grouping) {
             metadata = attribute.getMetadata();
         }
         for (Attribute<?> attr : in) {
-            if (attr.isToKeep()) {
+            if (attr.isToKeep() && !isOverloadedComposite) {
                 toKeep = true;
             }
             if (attr instanceof Attributes) {
                 Attributes attrs = (Attributes) attr;
                 int newAttributeCount = attrs.size();
-                int originalDataListSize = dataList.length;
+                int originalDataListSize = dataList.size();
                 if (newAttributeCount > 0) {
-                    if (dataList.length == 0) {
-                        dataList = new String[newAttributeCount];
-                        int i = 0;
+                    if (dataList.size() == 0) {
                         for (Attribute<?> a : attrs.getAttributes()) {
```

Comment:
you're not using Attribute<?> a in this for loop

Reply:
Ooops, I think that was a leftover from the original logic before refactor.

Reply:
Fixed!